File: Python/average_calculation.py
# from statistics import mean

# p63


def average_calculation(datas: list) -> dict:
    sum = 0
    count = 0
    average = 0.0

    for data in datas:
        sum += data
        count += 1

    if count != 0:
        average = sum/count

    return {"sum": sum, "count": count, "average": average}


datas = [54, 20, 88, 30, 71, 29, 44, 9]
result = average_calculation(datas)
print("合計値 = %d, 有効データ数 = %d, 平均値 = %f" % (
      result["sum"], result["count"], result["average"]))


# print("合計値 = %d, 有効データ数 = %d, 平均値 = %f" % (
#       sum(datas), len(datas), mean(datas)))

File: TypeScript/AverageCalculation.ts
// p63
const AverageCalculation = (datas: Array<number>) => {
    let sum: number = 0
    let count: number = 0
    let average: number = 0.0
    for (const data of datas) {
        sum += data
        count++
    }

    if (count !== 0) {
        average = sum / count
    }

    return { sum, count, average }
}

const datas = [54, 20, 88, 30, 71, 29, 44, 9]
const result = AverageCalculation(datas)

console.log(`合計値 = ${result.sum}, 有効データ数 = ${result.count}, 平均値 = ${result.average} `)

// const sum = datas.reduce((a, x) => { return a + x }, 0)
// console.log(`合計値 = ${sum}, 有効データ数 = ${datas.length}, 平均値 = ${sum / datas.length} `)